spec/tic_tac_toe_spec.rb

```diff
@@ -0,0 +1,206 @@
+def find_winner(board)
+    # returns row winner
+    board.each do |row|
+        row_winner = find_winner_in_line(row)
+        if row_winner
+            return row_winner
+        end
+    end
+
+    # returns column winner
+    board.transpose.each do |col|
+        col_winner = find_winner_in_line(col)
+        if col_winner
+            return col_winner
+        end
+    end
+
+    # returns diagonal winner
+    if find_winner_in_diagonal(board)
+        return find_winner_in_diagonal(board)
+    end
+
+    # check for any unfilled space
+    board.each do |row|
+        row.each do |col|
+            if col == '-'
+                return false
+            end
+        end
+    end
+
+    :tie
+end
+
+def find_winner_in_line(row)
+
+    if row.count('X') == 3
```

Comment:
This code assumes that the row will always have 3 elements, which is reasonable for a 3x3 tic tac toe game.  However, you might still find it useful to try calling `#all?` on the row.  It might look something like this:

```ruby
if row.all? { |cell| cell == 'X' }
  'X'
elsif row.all? { |cell| cell == 'O' }
  'O'
else
  false
end
```